### PR TITLE
Add overflow detection to `BigFloat#to_i64` and `#to_u64`

### PR DESCRIPTION
**Pull request by HertzDevil**
`BigFloat` values outside the representable range of `Int64` and `UInt64` can silently overflow:

```crystal
(2.0 ** 63 + 10000).to_big_f.to_i64 # => 10240
(2.0 ** 64 + 10000).to_big_f.to_u64 # => 8192
```

This PR adds overflow detection for these methods. The bounds are exact, and do not truncate `self` before conversion.

### CODE REVIEW

**Review by asterite — APPROVED** [review]

File: spec/std/big/big_float_spec.cr [hunk 1]
```diff
@@ -184,6 +184,18 @@ describe "BigFloat" do
     it { -4321.to_big_f.to_i!.should eq(-4321) }
   end
 
+  describe "to_i64" do
```
Comment (straight-shoota) [inline spec/std/big/big_float_spec.cr:187]:
```suggestion
  describe "#to_i64" do
```

Adding a method designation makes it easier to identify the specs.

File: spec/std/big/big_float_spec.cr [hunk 2]
```diff
@@ -184,6 +184,18 @@ describe "BigFloat" do
...
+    it { expect_raises(OverflowError) { (2.0 ** 63).to_big_f.to_i64 } }
+    it { expect_raises(OverflowError) { (BigFloat.new(2.0 ** 63, precision: 128) - 0.9999).to_i64 } }
+    it { expect_raises(OverflowError) { (-9.223372036854778e+18).to_big_f.to_i64 } } # (-(2.0 ** 63)).prev_float
+  end
+
+  describe "to_i64!" do
+    it { (2.0 ** 63).to_big_f.to_i64! }
```
Comment (straight-shoota) [inline spec/std/big/big_float_spec.cr:194]: Is it not possible to check the expected value for this conversion?

**Review by straight-shoota — APPROVED** [review]
While at it, it would be nice to add some positive expectations. Not strictly necessary for this PR, though.

File: spec/std/big/big_float_spec.cr (hunk 2 above)
Comment (HertzDevil) [reply spec/std/big/big_float_spec.cr:194]: The only expectation here is that it doesn't raise. Overflow is intentionally left undefined by GMP even if it returns a value.

Comment (straight-shoota) [reply spec/std/big/big_float_spec.cr:194]: Kay, maybe leave a comment about that?

**Review by straight-shoota — APPROVED** [review]

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**